lib/logger/lib/logger/config.ex

```diff
@@ -105,6 +105,16 @@ defmodule Logger.Config do
         {:ok, state}
 
       new_mode ->
+        if new_mode == :discard do
```

Comment:
The logs will be back by then, so we should be fine.

Reply:
We likely want to inform someone reading the logs that we stopped discarding. The order of the discard log messages will be confusing as it may appear logging resumed when it didn't because logs can arrive after the discard log message as they read `:async` or `:sync` just before it changed, and the message is sent after the discard log message. The only way to determine the end of a discarding period is to log it I think.

Reply:
Done.